client/analytics/settings/index.js

```diff
@@ -36,9 +36,19 @@ class Settings extends Component {
 		this.state = {
 			settings,
 			saving: false,
+			isDirty: false,
 		};
 
 		this.handleInputChange = this.handleInputChange.bind( this );
+		this.warnIfUnsavedChanges = this.warnIfUnsavedChanges.bind( this );
+	}
+
+	componentDidMount() {
+		window.addEventListener( 'beforeunload', this.warnIfUnsavedChanges );
```

Comment:
how about `history.listen()` for internal navigation?

https://github.com/ReactTraining/history

Reply:
So I just tried this out, and it appears to have the same issue that I was experiencing with `<Prompt />` i.e. if I hit cancel via a dialog thrown via `history.block()`, the address bar still updates. I'm guessing `Prompt` might just be wrapping this history module :)